pkg/resourcemanager/webhook/nodeagentauthorizer/authorizer.go

```diff
@@ -0,0 +1,234 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package nodeagentauthorizer
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	machinev1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	"github.com/go-logr/logr"
+	certificatesv1 "k8s.io/api/certificates/v1"
+	coordinationv1 "k8s.io/api/coordination/v1"
+	corev1 "k8s.io/api/core/v1"
+	eventsv1 "k8s.io/api/events/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	auth "k8s.io/apiserver/pkg/authorization/authorizer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+)
+
+// NewAuthorizer returns a new authorizer for requests from gardener-node-agents. It never has an opinion on the request.
+func NewAuthorizer(logger logr.Logger, sourceClient, targetClient client.Client, machineNamespace string) *authorizer {
+	return &authorizer{
+		sourceClient:     sourceClient,
+		targetClient:     targetClient,
+		logger:           logger,
+		machineNamespace: machineNamespace,
+	}
+}
+
+const valitailTokenSecretName = "gardener-valitail"
+
+var (
+	certificateSigningRequestResource = certificatesv1.Resource("certificatesigningrequests")
+	eventCoreResource                 = corev1.Resource("events")
+	eventResource                     = eventsv1.Resource("events")
+	leaseResource                     = coordinationv1.Resource("leases")
+	nodeResource                      = corev1.Resource("nodes")
+	secretsResource                   = corev1.Resource("secrets")
+)
+
+type authorizer struct {
+	sourceClient     client.Client
+	targetClient     client.Client
+	logger           logr.Logger
+	machineNamespace string
+}
+
+var _ = auth.Authorizer(&authorizer{})
```

Comment:
You can do this check without creating a pointer to a struct like this:
```suggestion
var _ auth.Authorizer = (*authorizer)(nil)
```